File: apzkr-pzpi-21-10-hotvianskyi-kyrylo/Task3-WebClient/src/interfaces/member.ts
import {PoolDto} from "./pool.ts";

export interface UpdateMemberCommand {
    firstName: string,
    lastName: string
}

export interface MemberDto {
    id: number,
    identityId : string,
    firstName: string,
    lastName: string,
    dateOfBirth: string,
    pool?: PoolDto
}

File: apzkr-pzpi-21-10-hotvianskyi-kyrylo/Task1-Server/Domain/Member.cs
﻿using Domain.Common;

namespace Domain;

public class Member : BaseEntity
{
    public string IdentityId { get; set; } = null!;
    public string FirstName { get; set; } = null!;
    public string LastName { get; set; } = null!;
    public string Sex { get; set; }
    public DateOnly DateOfBirth { get; set; }

    public int? PoolId { get; set; }
    public Pool? Pool { get; set; }
    public ICollection<Measurement>? Measurements { get; set; }
}